spring-modulith-junit/src/main/java/org/springframework/modulith/junit/TestExecutionCondition.java

```diff
@@ -59,7 +59,10 @@ ConditionEvaluationResult evaluate(ConditionContext context) {
 		}
 
 		if (!changes.hasClassChanges()) {
-			return enabled("No source file changes detected.");
+			return switch (changes.ON_NO_CHANGE) {
```

Comment:
Please move this into a `boolean skipOnNoChanges()` in `Changes`. This could then be used in a ternary `if` here.